src/main/scala/mesosphere/marathon/api/CORSFilter.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon.api
+
+import javax.inject.{ Inject }
+import javax.servlet._
+import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
+import mesosphere.marathon.{ MarathonConf }
+
+import scala.collection.JavaConverters._
+
+class CORSFilter @Inject() (config: MarathonConf) extends Filter {
+  // Map access_control_allow_origin flag into separate headers
+  lazy val origins: Seq[String] = config.accessControlAllowOrigin()
```

Comment:
I think you have to guard this on the condition `config.accessControlAllowOrigin.isSupplied`.